bundles/org.openhab.binding.insteon/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,453 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="insteon"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<bridge-type id="network">
+		<label>Insteon Network</label>
+		<description>An Insteon PLM or hub that is used to communicate with the Insteon devices.</description>
+
+		<config-description>
+			<parameter name="port" type="text" required="true">
+				<label>Port</label>
+				<description>Configuration information that is used to connect to PLM or hub.</description>
+			</parameter>
+
+			<parameter name="devicePollIntervalSeconds" type="integer" min="5" max="3600">
+				<label>Device Poll Interval</label>
+				<description>Device poll interval in seconds.</description>
+			</parameter>
+
+			<parameter name="modemDbRetryTimeoutSeconds" type="integer" min="30" max="600">
+				<label>Modem DB Retry Timeout</label>
+				<description>Modem DB retry timeout in seconds.</description>
+			</parameter>
+
+
+			<parameter name="additionalDevices" type="text">
+				<label>Additional Devices</label>
+				<description>Optional file with additional device types.</description>
+			</parameter>
+
+			<parameter name="additionalFeatures" type="text">
+				<label>Additional Features</label>
+				<description>Optional file with additional feature templates.</description>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="device">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="network" />
+		</supported-bridge-type-refs>
+
+		<label>Insteon Device</label>
+		<description>Insteon devices such as switches, dimmers, keypads, sensors, etc.</description>
+
+		<channels>
+			<channel id="acDelay" typeId="acDelay" />
```

Comment:
As far as I understand, every device just has a subset of those channels. In that case, you must only add the ones here that exist for ALL devices. All other channels can be added dynamically by the device handler upon initialisation.

Reply:
Yes that is correct. The list is compared with the ones defined for the specific device in device_types.xml. If it doesn't exist it is removed.  I'd rather leave it as is since it works. Is that OK?

Reply:
This might lead to buggy thing definitions if e.g. a thing is created manually or textually and it isn't immediately online. Furthermore, it can cause the framework to do a lot of unnecessary work as long as the list isn't updated.

What's the sense in adding everything and then removing channels? Codewise, adding the channels that exist for a specific device should be pretty much the same, shouldn't it? And it would avoid the situation where incorrect thing definitions exist.

Reply:
This is a bit frustrating, the documentation is lacking on how to do this. Now I have to refactor a bunch of code :(

Reply:
@kai, I'll make the changes it shouldn't be difficult.  It's just been frustrating when there are a bunch of unwritten rules that now need to be adhered to. Especially when the answer to questions is usually to look at other bindings and when you follow what they did, you are told you can't do that...

Reply:
removed, channels are not dynamically added vs culling unnecessary channels.

Reply:
>  It's just been frustrating when there are a bunch of unwritten rules that now need to be adhered to. 

Well, the unwritten rule is not to implement any bugs. And claiming that a thing has a channel that it doesn't have is imho a bug. It does not make sense to postulate any rules for that.

> to look at other bindings and when you follow what they did, you are told you can't do that

That's exactly the reason why I try to be strict in reviews, because any exception will be used against the reviewer in future :-). I personally am not aware of other bindings doing it that way and I'd recommend to all @openhab/add-ons-maintainers to also have a closer look at that in future. Can you tell where you have seen it? We can ask the maintainers to refactor those occurrences as well.

But anyhow: Thanks for adapting the code!

Reply:
When I implemented  the channels, I used the documentation at https://www.openhab.org/docs/developer/bindings/thing-xml.html#xml-structure-for-thing-descriptions. I guess I incorrectly assumed that I could put all channels in, and then remove the ones that aren't needed.

Reply:
BTW, I might be repeating myself, but if there is a set of things that @openhab/add-ons-maintainers look for when reviewing PR's it might be helpful to provide that checklist in advance to speed up reviews and eliminate frustration of the person doing the PR.